src/cephadm/cephadmlib/daemon_form.py

```diff
@@ -95,6 +95,26 @@ def __subclasshook__(cls, other: Type[DF]) -> bool:
         return callable(getattr(other, 'firewall_service_name', None))
 
 
+class UserGroupDaemonForm(DaemonForm, metaclass=abc.ABCMeta):
```

Comment:
we should check if possible to roll this into a bigger class since it's so specific

Reply:
I think it's justified because providing a good default is difficult and that the type based approach can find errors with mypy but not if we included it in ContainerDaemonBase with a default like `None`. However, I'm willing to continue discussing.